### PR TITLE
install latest intel igb driver 5.3.5.4 in the base image

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

File: src/igb/0001-add-support-for-BCM54616-phy-for-intel-igb-driver.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+diff --git a/src/e1000_82575.c b/src/e1000_82575.c
+index b4b973e..248c670 100644
+--- a/src/e1000_82575.c
++++ b/src/e1000_82575.c
+@@ -223,6 +223,7 @@ static s32 e1000_init_phy_params_82575(struct e1000_hw *hw)
+ 	case M88E1112_E_PHY_ID:
+ 	case M88E1340M_E_PHY_ID:
+ 	case M88E1111_I_PHY_ID:
++    case BCM54616_E_PHY_ID:
```
Comment (qiluo-msft) [inline src/igb/0001-add-support-for-BCM54616-phy-for-intel-igb-driver.patch]: >     [](start = 1, length = 4)

Use tab?

File: src/igb/0001-add-support-for-BCM54616-phy-for-intel-igb-driver.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ 		phy->ops.get_info	= e1000_get_phy_info_m88;
+diff --git a/src/e1000_defines.h b/src/e1000_defines.h
+index 6de3988..d5da148 100644
+--- a/src/e1000_defines.h
++++ b/src/e1000_defines.h
+@@ -1185,6 +1185,7 @@
+ #define I210_I_PHY_ID		0x01410C00
+ #define IGP04E1000_E_PHY_ID	0x02A80391
+ #define M88_VENDOR		0x0141
++#define BCM54616_E_PHY_ID   0x03625D10
```
Comment (qiluo-msft) [inline src/igb/0001-add-support-for-BCM54616-phy-for-intel-igb-driver.patch]: >    [](start = 26, length = 3)

use tab?

File: src/igb/Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.ONESHELL:
+SHELL = /bin/bash
+.SHELLFLAGS += -e
+
+MAIN_TARGET = igb.ko
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	rm -rf ./igb-$(IGB_DRIVER_VERSION)
+	wget -O igb-$(IGB_DRIVER_VERSION).tar.gz "http://downloads.sourceforge.net/project/e1000/igb%20stable/5.3.5.4/igb-5.3.5.4.tar.gz?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Fe1000%2Ffiles%2Figb%2520stable%2F5.3.5.4%2F&ts=1482880840&use_mirror=pilotfiber"
```
Comment (qiluo-msft) [inline src/igb/Makefile]: >http://downloads.sou [](start = 43, length = 20)

Use the official source: [https://downloadmirror.intel.com/13663/eng/igb-5.3.5.4.tar.gz](https://downloadmirror.intel.com/13663/eng/igb-5.3.5.4.tar.gz) ?
Or backup it to Azure Blob Storage and comment here with a sha1sum?

File: src/igb/Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	tar xzf igb-$(IGB_DRIVER_VERSION).tar.gz
+
+	# Patch
+	pushd ./igb-$(IGB_DRIVER_VERSION)
+	patch -p1 < ../0001-add-support-for-BCM54616-phy-for-intel-igb-driver.patch
+
+	# Build the package
+	pushd src
+	export BUILD_KERNEL=3.16.0-4-amd64
```
Comment (qiluo-msft) [inline src/igb/Makefile]: >3.16.0-4-amd64 [](start = 21, length = 14)

Use $KVERSION?

Comment (lguohan) [inline src/igb/Makefile]: BUILD_KERNEL is used by igb Makefile, will not change.

**Pull request merged**